python/cudf/cudf/core/udf/pipeline.py

```diff
@@ -219,23 +287,39 @@ def _kernel(retval, input_col_0, input_col_1, offset_0, offset_1, size):
         )
         row_initializers.append(row_initializer)
 
-    masked_input_initializers = "\n".join(initializers)
-    row_initializers = "\n".join(row_initializers)
-
     # Incorporate all of the above into the kernel code template
     d = {
         "input_columns": input_columns,
         "input_offsets": input_offsets,
-        "masked_input_initializers": masked_input_initializers,
-        "row_initializers": row_initializers,
+        "masked_input_initializers": "\n".join(initializers),
+        "row_initializers": "\n".join(row_initializers),
         "numba_rectype": row_type,  # from global
     }
 
     return kernel_template.format(**d)
 
 
+def _check_return_type(ty):
+    """
+    In almost every case, get_udf_return_type will throw a TypingError
+    if the user tries to use a field in the row containing a dtype that
+    is not supported. It does this by simply "not finding" overloads of
+    any operators for the ancillary dtype we type that field as. But it
+    is defeated by the following edge case:
+
+    def f(row):
+        return row[<bad dtype key>]
+
+    In this case numba is happy to return MaskedType(<bad dtype key>),
+    and won't throw because no operators were ever used. This function
+    explicitly checks for that case.
```

Comment:
```suggestion
    get_udf_return_type will throw a TypingError if the user tries to use
    a field in the row containing an unsupported dtype, except in the
    edge case where all the function does is return that element:
    
    def f(row):
        return row[<bad dtype key>]

    In this case numba is happy to return MaskedType(<bad dtype key>)
    because it relies on not finding overloaded operators for types to raise
    the exception, so we have to explicitly check for that case.
```

Reply:
Rolled with your comment :)